data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,236 @@
+    get_vm_guestnames_state_running=`virsh list --state-running | grep "${vm_guestnames_types}" | awk '{print $2}'`
+    vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})
+
+    while :
+    do
+        if [ -f ${MONITOR_LOCKED_FILE} ];then
+            #Check with LOCK-UP status
+            cat ${MONITOR_LOCKED_FILE} | grep "LOCKED"
+            if [ $? -eq 0 ];then
+                #KEEP LOCK-UP status
+                #To let just only one virsh console with one vm guest as backgroup PID
+
+                #wait for checking background PID of virsh console
+                sleep 3
+		#Check with background PID of virsh console
+		for vmguest in ${vm_guestnames_array[@]};do
+		    virsh list --state-running | grep ${vmguest}
+                    if [[ $? -eq 0 ]];then
+                        ps aux | grep "virsh console ${vmguest}" | grep -v grep
+                        if [[ $? -ne 0 ]];then
+                            get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+                        fi
+                    else
+                        virsh list --inactive | grep ${vmguest} | grep -v grep
+                        if [[ $? -eq 0 ]];then
+                            echo "${vmguest} is not running now"
+                            echo "No any output from virsh console ${vmguest}"
+                        fi
+                    fi
+                done
+            else
+                #Make sure UNLOCKED
+                echo "UNLOCK Monitor for $(basename $0)!"
+                exit			        
+            fi
+        else
+            echo "Create LOCK-UP for $(basename $0)"
+            echo "LOCKED" > ${MONITOR_LOCKED_FILE}
+            open_console
+        fi
+    done
+}
+
+function check_guest() {
+
+    #check with guest status
+    for vmguest in ${vm_guestnames_array[@]};do
+        echo -e ${vm_guestnames_inactive_array[*]} | grep ${vmguest} >> /dev/null 2>&1
+        if [[ $? -eq 0 ]];then
+            virsh start ${vmguest}
+            vmguest_failed=$((${vmguest_failed} | $(echo $?)))
+
+            #Quit if at least one vm guest failed to start up as normal
+            if [[ ${vmguest_failed} -ne 0 ]];then
+                echo "Fail to boot up ${vmguest} as normal. Please investigate.\n"
+                exit 1
+            fi
+        fi
+    done
+}
+
+##MAIN PART
+if [[ ! -d ${LOG_DIR} ]]; then
+     mkdir -p ${LOG_DIR}
+fi
+     
+if [[ ${START_MONITOR} -eq 1 ]];then
+    ##Install required packages
+    zypper_cmd="zypper --non-interactive in psmisc procps coreutils expect"
+    echo -e "${zypper_cmd} will be executed\n"
+    ${zypper_cmd} >> /dev/null 2>&1
+
+    #Check with the existed guest status
+    check_guest
+    
+    #Start Monitor fpr guest console
+    echo "Start Monitor in background for guest console"
+    start_monitor >> /dev/null 2>&1 &
+fi
+
+if [[ ${END_MONITOR} -eq 1 ]];then
+    end_monitor
+fi
+
+exit 0
```

Comment:
You do not have log file that records output of this script itself. You should have one, for example, guest_console_monitor_log and redirect output this file. Because while loop keeps running all the time, you should redirect output to this log file instead of print them on screen. Otherwise they will mess up with normal testing output.

Reply:
Great idea.
For now, all output from this script just only redirect to /dev/null do not print them on screen. 
Will update this.

Reply:
If you already redirect them to /dev/null, then I think you may do not need additional log file. It just record the loop. You can save it. I was just considering you forgot recording them. Agree to redirect to /dev/null.

Reply:
echo messages should also be redirected to /dev/null if you are not going to use log file.

Reply:
yes, for now, no plan to collect messages  and output from this script to use log file.

Reply:
You need to make sure they are not printed out on screen. Can not mess up with normal test output.

Reply:
Sure, got it. 
Quick updated, add monitor_log_file as guest_console_monitor.log.  

Collect all messages and output from this script to this monitor_log_file: 
For example: 
```
# cat /tmp/virt_logs_residence/guest_console_monitor.log 
zypper --non-interactive in psmisc procps coreutils expect will be executed

Start Monitor in background for guest console

Create LOCK-UP for guest_console_monitor.sh

End of Monitor for guest console

kill PIDs guest_console_monitor.sh

Kill all associated virsh console processes
```


Also, Changed the output as below: 
->   Start Monitor for guest console
```
# ./guest_console_monitor.sh -s
Start Monitor in background for guest console
```
-> End of Monitor for guest console
```
# ./guest_console_monitor.sh -e
End of Monitor for guest console
```

Reply:
DONE